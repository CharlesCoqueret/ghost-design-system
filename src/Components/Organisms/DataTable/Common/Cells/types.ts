import {
  IColumnType,
  IExtraEditableDataTableProps,
  IExtraLineEditableInPlaceDataTableProps,
  IExtraStaticDataTableProps,
} from '../types';

export interface ICellProps<T, U extends IColumnType<T>> {
  column: U;
  dataTestId?: string;
  editing?: boolean;
  extra?: IExtraStaticDataTableProps<T> | IExtraLineEditableInPlaceDataTableProps<T> | IExtraEditableDataTableProps<T>;
  forcedValue?: T[keyof T] | number | string | Date | null;
  onChange?: (newValue: T[keyof T]) => void;
  row?: T;
  rowIndex: number;
}
