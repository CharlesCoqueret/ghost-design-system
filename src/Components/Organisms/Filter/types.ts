import { ThousandsGroupStyle } from '../../Atoms/AmountInput';
import { DateFormat, WeekDayEnum } from '../../Atoms/DatePickerInput';
import { IOption } from '../../Atoms/SelectInput';

export enum FilterTypeEnum {
  CHECKBOX = 'checkbox',
  DATE = 'date',
  DYNAMICSEARCH = 'dynamicsearch',
  MULTISELECT = 'multiselect',
  NUMBER = 'number',
  COLUMN = 'column',
  SELECT = 'select',
  TEXT = 'text',
  TITLE = 'title',
}

export type IFilterLayoutAndFieldsProps<T> = IFilerColProps<T> | IFilerTitleProps | IFilterFieldsProps<T>;

export type IFilterFieldsProps<T> =
  | IFilterCheckboxProps<T>
  | IFilterDateProps<T>
  | IFilterDynamicSearchProps<T>
  | IFilterMultiSelectProps<T>
  | IFilterNumberProps<T>
  | IFilterSelectProps<T>
  | IFilterTextProps<T>;

export interface IFilterBaseProps<T> {
  dataIndex: keyof T;
  dataTestId?: string;
  fieldClassName?: string;
  fieldSize?: number;
  helperText?: string;
  inputClassName?: string;
  label?: string;
}

export interface IFilterCheckboxProps<T> extends IFilterBaseProps<T> {
  filterType: FilterTypeEnum.CHECKBOX;
}

export interface IFilterDateProps<T> extends IFilterBaseProps<T> {
  calendarStartDay?: WeekDayEnum;
  dateFormat?: DateFormat;
  filterType: FilterTypeEnum.DATE;
  isClearable?: boolean;
  locale?: string;
  name?: string;
  placeholder?: string;
}

export interface IFilterDynamicSearchProps<T> extends IFilterBaseProps<T> {
  colors?: {
    controlErrorColor: string; // colors.error,
    controlFocusColor: string; // colors.primary,
    fontColor: string; // 'rgb(0, 0, 0)',
    optionFocusColor: string; // colors.chalk,
    optionSelectedColor: string; // colors.primary,
  };
  filterType: FilterTypeEnum.DYNAMICSEARCH;
  isClearable?: boolean;
  name?: string;
  noOptionsMessage: (obj: { inputValue: string }) => string;
  placeholder?: string;
  resolveValue: (value: string | number) => Promise<IOption | undefined>;
  searchOptions: (searchTerm: string) => Promise<Array<IOption> | undefined>;
}

export interface IFilterMultiSelectProps<T> extends IFilterBaseProps<T> {
  colors?: {
    controlErrorColor: string; // colors.error,
    controlFocusColor: string; // colors.primary,
    fontColor: string; // 'rgb(0, 0, 0)',
    optionFocusColor: string; // colors.chalk,
    optionSelectedColor: string; // colors.primary,
  };
  filterType: FilterTypeEnum.MULTISELECT;
  isClearable?: boolean;
  name?: string;
  numberOfItemLabel: string;
  numberOfItemsLabel: string;
  options: Array<IOption>;
  placeholder?: string;
}

export interface IFilterNumberProps<T> extends IFilterBaseProps<T> {
  allowNegative?: boolean;
  decimalScale?: number;
  decimalSeparator?: string;
  filterType: FilterTypeEnum.NUMBER;
  maxValue?: number;
  minValue?: number;
  name?: string;
  placeholder?: string;
  prefix?: string | undefined;
  suffix?: string | undefined;
  thousandSeparator?: string;
  thousandsGroupStyle?: ThousandsGroupStyle;
}

export interface IFilerColProps<T> {
  filterType: FilterTypeEnum.COLUMN;
  fields: Array<IFilterLayoutAndFieldsProps<T>>;
}

export interface IFilterSelectProps<T> extends IFilterBaseProps<T> {
  colors?: {
    controlErrorColor: string; // colors.error,
    controlFocusColor: string; // colors.primary,
    fontColor: string; // 'rgb(0, 0, 0)',
    optionFocusColor: string; // colors.chalk,
    optionSelectedColor: string; // colors.primary,
  };
  filterType: FilterTypeEnum.SELECT;
  isClearable?: boolean;
  name?: string;
  options: Array<IOption>;
  placeholder?: string;
}

export interface IFilterTextProps<T> extends IFilterBaseProps<T> {
  filterType: FilterTypeEnum.TEXT;
  maxLength?: number;
  minLength?: number;
  name?: string;
  placeholder?: string;
}

export interface IFilerTitleProps {
  filterType: FilterTypeEnum.TITLE;
  label: string;
}
