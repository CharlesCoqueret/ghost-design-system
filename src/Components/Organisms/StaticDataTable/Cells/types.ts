import { IColumnType, IExtraLineEditableDataTableProps, IExtraStaticDataTableProps } from '../types';

export interface ICellProps<T, U extends IColumnType<T>> {
  column: U;
  row?: T;
  rowIndex: number;
  extra?: IExtraStaticDataTableProps<T> | IExtraLineEditableDataTableProps<T>;
  forcedValue?: T[keyof T] | number | string | Date | null;
  onChange?: (newValue?: T[keyof T]) => void;
}
