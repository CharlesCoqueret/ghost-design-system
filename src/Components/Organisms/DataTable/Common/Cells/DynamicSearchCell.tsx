import React, { ReactElement } from 'react';
import classnames from 'classnames';

import { ICellProps } from './types';
import { IColumnDynamicSearch } from '../types';
import { DynamicSearchField } from '../../../../Molecules/DynamicSearchField';

const DynamicSearchCell = <T,>(props: ICellProps<T, IColumnDynamicSearch<T>>): ReactElement => {
  const { column, dataTestId, editing, extra, forcedValue, onChange, row, rowIndex } = props;

  const displayValue = (forcedValue || (row && row[column.dataIndex]) || undefined) as string | undefined;
  const isCurrentlyEditedRow =
    editing || (extra && 'editedRowIndex' in extra ? extra.editedRowIndex === rowIndex && column.editable : false);

  return (
    <td className={classnames({ ellipsis: column.ellipsis })} style={{ display: column.hidden ? 'none' : undefined }}>
      <DynamicSearchField
        colors={column.colors}
        dataTestId={dataTestId}
        ellipsis={column.ellipsis}
        inputValue={displayValue}
        isClearable={column.isClearable}
        name={column.title.toString()}
        noOptionsMessage={column.noOptionsMessage}
        onChange={(newValue: string | number | null | undefined) => {
          if (onChange) {
            onChange(newValue as unknown as T[keyof T]);
          }
        }}
        searchOptions={column.searchOptions}
        resolveValue={column.resolveValue}
        readOnly={!isCurrentlyEditedRow}
      />
    </td>
  );
};

export default DynamicSearchCell;
