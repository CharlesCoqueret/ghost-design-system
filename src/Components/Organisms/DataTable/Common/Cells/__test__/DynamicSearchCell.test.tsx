import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import DynamicSearchCell from '../DynamicSearchCell';
import { ColumnType } from '../../types';

describe('DynamicSearchCell component', () => {
  it('DynamicSearchCell renders', () => {
    const resolveValueMock = jest.fn();
    const searchOptionsMock = jest.fn();
    const noOptionsMessageMock = jest.fn();

    const { container } = render(
      <table>
        <tbody>
          <tr>
            <DynamicSearchCell
              column={{
                dataIndex: 'data',
                noOptionsMessage: noOptionsMessageMock,
                resolveValue: resolveValueMock,
                searchOptions: searchOptionsMock,
                title: 'DynamicSearchCell',
                type: ColumnType.DYNAMICSEARCH,
              }}
              rowIndex={0}
            />
          </tr>
        </tbody>
      </table>,
    );

    expect(container).toMatchSnapshot();
    expect(resolveValueMock).toBeCalledTimes(0);
    expect(searchOptionsMock).toBeCalledTimes(0);
  });

  it('DynamicSearchCell renders with forced value', async () => {
    const resolveValueMock = jest.fn().mockImplementation(() => {
      return Promise.resolve({ value: 'value2', label: 'Label 2' });
    });
    const searchOptionsMock = jest.fn();
    const noOptionsMessageMock = jest.fn();

    const { container } = render(
      <table>
        <tbody>
          <tr>
            <DynamicSearchCell
              column={{
                dataIndex: 'data',
                noOptionsMessage: noOptionsMessageMock,
                resolveValue: resolveValueMock,
                searchOptions: searchOptionsMock,
                title: 'DynamicSearchCell',
                type: ColumnType.DYNAMICSEARCH,
              }}
              dataTestId='DATA-TEST-ID'
              forcedValue='value2'
              rowIndex={0}
            />
          </tr>
        </tbody>
      </table>,
    );

    expect(await screen.findByTestId('DATA-TEST-ID-spinner')).toBeTruthy();

    expect(screen.queryByTestId('DATA-TEST-ID-spinner')).toBeFalsy();

    expect(container).toMatchSnapshot();
    expect(resolveValueMock).toBeCalledTimes(1);
    expect(resolveValueMock).toBeCalledWith('value2');
    expect(searchOptionsMock).toBeCalledTimes(0);
  });

  it('DynamicSearchCell renders when hidden', () => {
    const resolveValueMock = jest.fn();
    const searchOptionsMock = jest.fn();
    const noOptionsMessageMock = jest.fn();

    const { container } = render(
      <table>
        <tbody>
          <tr>
            <DynamicSearchCell
              column={{
                dataIndex: 'data',
                hidden: true,
                noOptionsMessage: noOptionsMessageMock,
                resolveValue: resolveValueMock,
                searchOptions: searchOptionsMock,
                title: 'DynamicSearchCell',
                type: ColumnType.DYNAMICSEARCH,
              }}
              rowIndex={0}
            />
          </tr>
        </tbody>
      </table>,
    );

    expect(container).toMatchSnapshot();
  });

  it('DynamicSearchCell renders in edit mode and handles change', async () => {
    const onChangeMock = jest.fn();

    const resolveValueMock = jest.fn().mockImplementation(() => {
      return Promise.resolve({ value: 'value1', label: 'Label 1' });
    });
    const searchOptionsMock = jest.fn().mockImplementation(() => {
      return Promise.resolve([
        { value: 'value1', label: 'Label 1' },
        { value: 'value2', label: 'Label 2' },
      ]);
    });
    const noOptionsMessageMock = jest.fn();

    const { container } = render(
      <table>
        <tbody>
          <tr>
            <DynamicSearchCell
              column={{
                dataIndex: 'data',
                isClearable: true,
                noOptionsMessage: noOptionsMessageMock,
                resolveValue: resolveValueMock,
                searchOptions: searchOptionsMock,
                title: 'DynamicSearchCell',
                type: ColumnType.DYNAMICSEARCH,
              }}
              dataTestId='DATA-TEST-ID'
              editing
              onChange={onChangeMock}
              row={{ data: 'value1' }}
              rowIndex={0}
            />
          </tr>
        </tbody>
      </table>,
    );

    expect(await screen.findByTestId('DATA-TEST-ID-spinner')).toBeTruthy();

    expect(screen.queryByTestId('DATA-TEST-ID-spinner')).toBeFalsy();

    expect(container).toMatchSnapshot();

    const select = await screen.findByRole('combobox');
    userEvent.type(select, '{backspace}');

    expect(container).toMatchSnapshot();
    expect(onChangeMock).toBeCalledTimes(1);
    expect(onChangeMock).toBeCalledWith(undefined);
  });

  it('DynamicSearchCell handles display of no message', async () => {
    const onChangeMock = jest.fn();

    const resolveValueMock = jest.fn().mockImplementation(() => {
      return Promise.resolve({ value: 'value1', label: 'Label 1' });
    });
    const searchOptionsMock = jest.fn().mockImplementation(() => {
      return Promise.resolve(undefined);
    });
    const noOptionsMessageMock = jest.fn().mockImplementation((obj: { inputValue: string }) => {
      return `No result for ${obj.inputValue}`;
    });

    const { container } = render(
      <table>
        <tbody>
          <tr>
            <DynamicSearchCell
              column={{
                dataIndex: 'data',
                isClearable: true,
                noOptionsMessage: noOptionsMessageMock,
                resolveValue: resolveValueMock,
                searchOptions: searchOptionsMock,
                title: 'DynamicSearchCell',
                type: ColumnType.DYNAMICSEARCH,
              }}
              dataTestId='DATA-TEST-ID'
              editing
              onChange={onChangeMock}
              row={{ data: 'value1' }}
              rowIndex={0}
            />
          </tr>
        </tbody>
      </table>,
    );

    expect(await screen.findByTestId('DATA-TEST-ID-spinner')).toBeTruthy();

    expect(screen.queryByTestId('DATA-TEST-ID-spinner')).toBeFalsy();

    expect(container).toMatchSnapshot();

    const select = await screen.findByRole('combobox');
    userEvent.type(select, '{backspace}');

    expect(container).toMatchSnapshot();
    expect(noOptionsMessageMock).toBeCalledTimes(2);
    expect(onChangeMock).toBeCalledTimes(1);
    expect(onChangeMock).toBeCalledWith(undefined);
  });

  it('DynamicSearchCell renders in edit mode via extra', () => {
    const resolveValueMock = jest.fn();
    const searchOptionsMock = jest.fn();
    const noOptionsMessageMock = jest.fn();

    const { container } = render(
      <table>
        <tbody>
          <tr>
            <DynamicSearchCell
              column={{
                dataIndex: 'data',
                hidden: true,
                noOptionsMessage: noOptionsMessageMock,
                resolveValue: resolveValueMock,
                searchOptions: searchOptionsMock,
                title: 'DynamicSearchCell',
                type: ColumnType.DYNAMICSEARCH,
              }}
              extra={{ editedRowIndex: 0 }}
              rowIndex={0}
            />
          </tr>
        </tbody>
      </table>,
    );

    expect(container).toMatchSnapshot();
  });
});
