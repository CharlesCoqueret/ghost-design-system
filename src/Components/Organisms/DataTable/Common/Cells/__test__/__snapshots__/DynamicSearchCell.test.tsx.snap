// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicSearchCell component DynamicSearchCell handles display of no message 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          class=""
        >
          <div
            class="gds-field-group"
          >
            <div
              class="field gds-select-container input-select-field input-select-field"
            >
              <div
                class=" css-1959ul7-container"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-3-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class=" css-1l9db9l-control"
                >
                  <div
                    class=" css-9plg90-ValueContainer"
                  >
                    <div
                      class=" css-qc6sy-singleValue"
                    >
                      Label 1
                    </div>
                    <div
                      class=" css-6j8wv5-Input"
                      data-value=""
                    >
                      <input
                        aria-autocomplete="list"
                        aria-expanded="false"
                        aria-haspopup="true"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class=""
                        id="DATA-TEST-ID"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class=" css-1hb7zxy-IndicatorsContainer"
                  >
                    <div
                      aria-hidden="true"
                    >
                      <i
                        class="dynamic-search-icon icon"
                        data-testid="DATA-TEST-ID-clear"
                        icon="fal,xmark"
                      />
                    </div>
                    <span
                      class=" css-1okebmr-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                    >
                      <i
                        class="dynamic-search-icon icon"
                        data-testid="DATA-TEST-ID-magnifier"
                        icon="fal,magnifying-glass"
                      />
                    </div>
                  </div>
                </div>
                <input
                  name="DynamicSearchCell"
                  type="hidden"
                  value="value1"
                />
              </div>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`DynamicSearchCell component DynamicSearchCell handles display of no message 2`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          class=""
        >
          <div
            class="gds-field-group"
          >
            <div
              class="field gds-select-container input-select-field input-select-field"
            >
              <div
                class=" css-1959ul7-container"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-3-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                >
                  <span
                    id="aria-selection"
                  >
                    All selected options have been cleared.
                  </span>
                  <span
                    id="aria-context"
                  >
                      Use Up and Down to choose options, press Enter to select the currently focused option, press Escape to exit the menu, press Tab to select the option and exit the menu.
                  </span>
                </span>
                <div
                  class=" css-1tkmbhg-control"
                >
                  <div
                    class=" css-9plg90-ValueContainer"
                  >
                    <div
                      class=" css-14el2xx-placeholder"
                      id="react-select-3-placeholder"
                    >
                      Select...
                    </div>
                    <div
                      class=" css-6j8wv5-Input"
                      data-value=""
                    >
                      <input
                        aria-autocomplete="list"
                        aria-controls="react-select-3-listbox"
                        aria-describedby="react-select-3-placeholder"
                        aria-expanded="true"
                        aria-haspopup="true"
                        aria-owns="react-select-3-listbox"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class=""
                        id="DATA-TEST-ID"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class=" css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class=" css-1okebmr-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                    >
                      <i
                        class="dynamic-search-icon icon"
                        data-testid="DATA-TEST-ID-magnifier"
                        icon="fal,magnifying-glass"
                      />
                    </div>
                  </div>
                </div>
                <div>
                  <div
                    class=" css-omztxt"
                  >
                    <div
                      class=" css-26l3qy-menu"
                      id="react-select-3-listbox"
                    >
                      <div
                        class="css-bp8cua-ScrollManager"
                      />
                      <div
                        class=" css-4ljt47-MenuList"
                      >
                        <div
                          class=" css-gg45go-NoOptionsMessage"
                        >
                          No result for 
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <input
                  name="DynamicSearchCell"
                  type="hidden"
                  value=""
                />
              </div>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`DynamicSearchCell component DynamicSearchCell renders 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          class=""
        >
          <div
            class="gds-field-group"
          >
            <div
              class="field gds-select-container input-select-field-read-only input-select-field"
            >
              -
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`DynamicSearchCell component DynamicSearchCell renders in edit mode and handles change 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          class=""
        >
          <div
            class="gds-field-group"
          >
            <div
              class="field gds-select-container input-select-field input-select-field"
            >
              <div
                class=" css-1959ul7-container"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-2-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class=" css-1l9db9l-control"
                >
                  <div
                    class=" css-9plg90-ValueContainer"
                  >
                    <div
                      class=" css-qc6sy-singleValue"
                    >
                      Label 1
                    </div>
                    <div
                      class=" css-6j8wv5-Input"
                      data-value=""
                    >
                      <input
                        aria-autocomplete="list"
                        aria-expanded="false"
                        aria-haspopup="true"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class=""
                        id="DATA-TEST-ID"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class=" css-1hb7zxy-IndicatorsContainer"
                  >
                    <div
                      aria-hidden="true"
                    >
                      <i
                        class="dynamic-search-icon icon"
                        data-testid="DATA-TEST-ID-clear"
                        icon="fal,xmark"
                      />
                    </div>
                    <span
                      class=" css-1okebmr-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                    >
                      <i
                        class="dynamic-search-icon icon"
                        data-testid="DATA-TEST-ID-magnifier"
                        icon="fal,magnifying-glass"
                      />
                    </div>
                  </div>
                </div>
                <input
                  name="DynamicSearchCell"
                  type="hidden"
                  value="value1"
                />
              </div>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`DynamicSearchCell component DynamicSearchCell renders in edit mode and handles change 2`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          class=""
        >
          <div
            class="gds-field-group"
          >
            <div
              class="field gds-select-container input-select-field input-select-field"
            >
              <div
                class=" css-1959ul7-container"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-2-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                >
                  <span
                    id="aria-selection"
                  >
                    All selected options have been cleared.
                  </span>
                  <span
                    id="aria-context"
                  >
                    option Label 1 focused, 1 of 2. 2 results available. Use Up and Down to choose options, press Enter to select the currently focused option, press Escape to exit the menu, press Tab to select the option and exit the menu.
                  </span>
                </span>
                <div
                  class=" css-1tkmbhg-control"
                >
                  <div
                    class=" css-9plg90-ValueContainer"
                  >
                    <div
                      class=" css-14el2xx-placeholder"
                      id="react-select-2-placeholder"
                    >
                      Select...
                    </div>
                    <div
                      class=" css-6j8wv5-Input"
                      data-value=""
                    >
                      <input
                        aria-autocomplete="list"
                        aria-controls="react-select-2-listbox"
                        aria-describedby="react-select-2-placeholder"
                        aria-expanded="true"
                        aria-haspopup="true"
                        aria-owns="react-select-2-listbox"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class=""
                        id="DATA-TEST-ID"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class=" css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class=" css-1okebmr-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                    >
                      <i
                        class="dynamic-search-icon icon"
                        data-testid="DATA-TEST-ID-magnifier"
                        icon="fal,magnifying-glass"
                      />
                    </div>
                  </div>
                </div>
                <div>
                  <div
                    class=" css-omztxt"
                  >
                    <div
                      class=" css-26l3qy-menu"
                      id="react-select-2-listbox"
                    >
                      <div
                        class="css-bp8cua-ScrollManager"
                      />
                      <div
                        class=" css-4ljt47-MenuList"
                      >
                        <div
                          aria-disabled="false"
                          class=" css-cfsx9x-option"
                          id="react-select-2-option-0"
                          tabindex="-1"
                        >
                          Label 1
                        </div>
                        <div
                          aria-disabled="false"
                          class=" css-pgxi80-option"
                          id="react-select-2-option-1"
                          tabindex="-1"
                        >
                          Label 2
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <input
                  name="DynamicSearchCell"
                  type="hidden"
                  value=""
                />
              </div>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`DynamicSearchCell component DynamicSearchCell renders in edit mode via extra 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          class=""
          style="display: none;"
        >
          <div
            class="gds-field-group"
          >
            <div
              class="field gds-select-container input-select-field input-select-field"
            >
              <div
                class=" css-1959ul7-container"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-4-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class=" css-1l9db9l-control"
                >
                  <div
                    class=" css-9plg90-ValueContainer"
                  >
                    <div
                      class=" css-14el2xx-placeholder"
                      id="react-select-4-placeholder"
                    >
                      Select...
                    </div>
                    <div
                      class=" css-6j8wv5-Input"
                      data-value=""
                    >
                      <input
                        aria-autocomplete="list"
                        aria-describedby="react-select-4-placeholder"
                        aria-expanded="false"
                        aria-haspopup="true"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class=""
                        id="react-select-4-input"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class=" css-1hb7zxy-IndicatorsContainer"
                  >
                    <div
                      aria-hidden="true"
                    >
                      <svg
                        aria-hidden="true"
                        class="icon gds-loader size-1x dynamic-search-spinner"
                        focusable="false"
                        height="50"
                        icon="fal,spinner"
                        preserveAspectRatio="xMidYMid"
                        role="img"
                        style="background: transparent;"
                        viewBox="0 0 100 100"
                        width="50"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <circle
                          cx="50"
                          cy="50"
                          fill="transparent"
                          r="30"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-width="10"
                        >
                          <animatetransform
                            attributeName="transform"
                            dur="2s"
                            keyTimes="0;0.5;1"
                            repeatCount="indefinite"
                            type="rotate"
                            values="0 50 50;180 50 50;720 50 50"
                          />
                          <animate
                            attributeName="stroke-dasharray"
                            dur="2s"
                            keyTimes="0;0.5;1"
                            repeatCount="indefinite"
                            values="20 150; 150 20; 20 150"
                          />
                        </circle>
                      </svg>
                    </div>
                    <span
                      class=" css-1okebmr-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                    >
                      <i
                        class="dynamic-search-icon icon"
                        icon="fal,magnifying-glass"
                      />
                    </div>
                  </div>
                </div>
                <input
                  name="DynamicSearchCell"
                  type="hidden"
                  value=""
                />
              </div>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`DynamicSearchCell component DynamicSearchCell renders when hidden 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          class=""
          style="display: none;"
        >
          <div
            class="gds-field-group"
          >
            <div
              class="field gds-select-container input-select-field-read-only input-select-field"
            >
              -
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`DynamicSearchCell component DynamicSearchCell renders with forced value 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          class=""
        >
          <div
            class="gds-field-group"
          >
            <div
              class="field gds-select-container input-select-field-read-only input-select-field"
              data-testid="DATA-TEST-ID"
            >
              <span
                class="gds-typography"
              >
                Label 2
              </span>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
